gesturelibrary/src/gesturelib.c

```diff
@@ -91,15 +98,13 @@ unsigned int assign_group(touch_event_t* touch_event) {
         return TOUCH_ID_UNDEFINED;
     }
 
-    touch_event->id             = closest_group;
-    groups_heads[closest_group] = touch_event;
-
+    touch_event->id = closest_group;
     return closest_group;
 }
 
-static float squared_distance(touch_event_t* a, touch_event_t* b) {
-    float dx = a->position_x - b->position_x;
-    float dy = a->position_y - b->position_y;
+static float squared_distance(touch_event_t a, touch_event_t b) {
```

Comment:
Why are you using copies instead of pointers?
If you are worried about modification, use `const touch_event*`

Reply:
Fair point. Updated